### PR TITLE
Don't use implicit string conversions

### PR DESCRIPTION
**Pull request by luis-pereira**
Make us aware of the implicit "casts".

### CODE REVIEW

**Comment by tsujan** [comment]
Yes, of course, it should be done, like in other parts of LXQt. One of us will have to rebase later because I want to add a PR; I hope it will be you ;)

File: src/foldermodel.cpp [hunk 1]
```diff
@@ -224,7 +224,7 @@ QString FolderModel::makeTooltip(FolderModelItem* item) const {
         tip += QStringLiteral("<p><i>") + tr("Location:") + QStringLiteral("</i> ") + QString::fromUtf8(parent_str.get()) + QStringLiteral("</p>");
     }
     // file type
-    tip += QStringLiteral("<i>") + tr("File type:") + QStringLiteral("</i> ") + info->mimeType()->desc();
+    tip += QStringLiteral("<i>") + tr("File type:") + QStringLiteral("</i> ") + QLatin1String(info->mimeType()->desc());
```
Comment (tsujan) [inline src/foldermodel.cpp]: Wouldn't `QString::fromUtf8(info->mimeType()->desc())` be more consistent with other changes?

File: src/foldermodel.cpp [hunk 2]
```diff
@@ -264,7 +264,7 @@ QVariant FolderModel::data(const QModelIndex& index, int role/* = Qt::DisplayRol
             return (showFullNames_ && !item->name().empty() ? QString::fromStdString(item->name())
                                                             : item->displayName());
         case ColumnFileType:
-            return QString(info->mimeType()->desc());
+            return QString::fromUtf8((info->mimeType()->desc()));
```
Comment (tsujan) [inline src/foldermodel.cpp]: Has an extra pair pf parentheses.

File: src/foldermodel.cpp (hunk 1 above)
Comment (luis-pereira) [reply src/foldermodel.cpp]: Yes, it should. That one escaped.

File: src/foldermodel.cpp (hunk 2 above)
Comment (luis-pereira) [reply src/foldermodel.cpp]: nice catch

**Comment by luis-pereira** [comment]
It's already done in a big part of LXQt.

File: src/filedialog.cpp [hunk 3]
```diff
@@ -969,17 +969,17 @@ void FileDialog::FileDialogFilter::update() {
...
             right = nameFilter.length();
         }
         nameFilter = nameFilter.mid(left, right - left);
     }
     // parse the "*.ext1 *.ext2 *.ext3 ..." list into QRegularExpression objects
-    auto globs = nameFilter.simplified().split(' ');
+    auto globs = nameFilter.simplified().split(QLatin1Char(' '));
     for(const auto& glob: globs) {
 #if (QT_VERSION >= QT_VERSION_CHECK(5,12,0))
         patterns_.emplace_back(QRegularExpression("\\A(?:"
```
[the remark this answers is not in the document]
Comment (luis-pereira) [reply src/filedialog.cpp]: Using 5.11 here. will update.

**Pull request merged**